### ISSUE TITLE
test of unifile 1.0.0 

### ISSUE THREAD

**Issue by lexoyo** [silexlabs/unifile#62]
@JbIPS great work on the 1.0.0 version of unifile
https://github.com/JbIPS/unifile/tree/1.0.0

When I do this I have errors like these

```
$ npm i && npm test

  1) Unifile class constructor create a new instance with config:
     AssertionError: expected undefined to deeply equal { key: 'value' }
      at Assertion.assertEqual (node_modules/chai/lib/chai/core/assertions.js:485:19)
      at Assertion.ctx.(anonymous function) [as equal] (node_modules/chai/lib/chai/utils/addMethod.js:41:
25)
      at Context.<anonymous> (test/index.js:26:38)

  2) Unifile class getAuthorizeURL() throws an error if connectorName is undefined:
     AssertionError: expected [Function] to throw error matching /Unknown connector/ but got 'Cannot read
 property \'toLowerCase\' of undefined'
      at Context.<anonymous> (test/index.js:61:26)
```

Are the tests up to date, did I miss something? I have node v6.2.2...

Also I tried

```
$ cd samples
$ npm i
$ node simple-api-server.js 
Listening on 6805
```

Then I open `http://localhost:6805/` and click on "Login", but it opens a popup on `http://localhost:6805/github/oauth-callback?code=a25b7f4b4e268321dcad&state=aaathub` and says "Cannot GET /github/oauth-callback?code=a25b7f4b4e268321dcad&state=aaathub"

Finally, how can I play with unifile and the command line? :seedling:

**Comment by JbIPS** [comment]
I forgot the tests along the development line :smile:, I'll update them soon.

The 404 error is weird, I'll have a look.

Finally, if you wanna try the CLI, you can play with [Unifile-CLI](https://github.com/silexlabs/unifile-cli). Have fun!

**Comment by lexoyo** [comment]
bump
I am starting to develop a front end... I need an example of the json output

**Comment by JbIPS** [comment]
The file representation for now is:
```
{
  bytes: Number,
  modified: Date,
  name: String,
  is_dir: Bool
}
```
Reading a folder will return an Array of file.

BTW, what do you think of this structure? I kept it from previous version but I'm not sure it's enough/accurate. For example, I dislike underscore in variables names so I'll surely change it to `isDir`.

**Comment by JbIPS** [comment]
Alright, I fixed the units test and try to reproduce your sample error but it works well with the latest version. Could you give it another try and tell me if you still have this error? Thanks

**Comment by lexoyo** [comment]
unit tests are fixed :+1: 

the error I had was because the default app on GitHub is redirecting to localhost:6805 and I was using a different port :+1: 

I plugged my prototype of cloud explorer on the new unifile, and `ls` works fine
https://github.com/lexoyo/CloudExplorer2/tree/with_unifile_API

It is on a branch for now until I have tested it more

This is really good! It could be even better with
* a doc about the apps (github, dropbox, ...) which I should have created
* the link between unifile and CE is `simple-api-server.js` which I have duplicated in CE and adapted to my needs, i.e. I changed the folder served on `/` which was the test page - this whole thing may have to be cleaned up?
* the default app on Dropbox is not declared as "production" in Dropbox, so it is not usable and says:

![screenshot from 2016-07-16 01-18-33](https://cloud.githubusercontent.com/assets/715377/16890881/3db364e8-4af3-11e6-97ea-29887a88e359.png)
